File: app/src/main/java/com/kidozh/discuzhub/results/HotForumsResult.java
package com.kidozh.discuzhub.results;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.kidozh.discuzhub.entities.HotForum;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class HotForumsResult extends BaseResult {
    @JsonProperty("Variables")
    public HotForumVariables variables;

    public static class HotForumVariables extends VariableResults{


        @JsonProperty("data")
        public List<HotForum> hotForumList;

    }

    public boolean isError(){
        return this.message != null;
    }
}

File: app/src/main/java/com/kidozh/discuzhub/results/HotForumsResult.kt
package com.kidozh.discuzhub.results

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import com.kidozh.discuzhub.entities.HotForum

@JsonIgnoreProperties(ignoreUnknown = true)
class HotForumsResult : BaseResult() {
    @JvmField
    @JsonProperty("Variables")
    var variables: HotForumVariables = HotForumVariables()

    class HotForumVariables : VariableResults() {
        @JvmField
        @JsonProperty("data")
        var hotForumList: List<HotForum> = ArrayList()
    }

    override fun isError(): Boolean {
        return message != null
    }
}